test/unit/algo/array_utils.cpp

```diff
@@ -0,0 +1,42 @@
+//==================================================================================================
+/**
+  EVE - Expressive Vector Engine
+  Copyright : EVE Contributors & Maintainers
+  SPDX-License-Identifier: MIT
+**/
+//==================================================================================================
+
+#include "test.hpp"
+
+#include <eve/algo/array_utils.hpp>
+
+#include <algorithm>
+#include <functional>
+
+TTS_CASE("eve.algo.array_utils array_map")
+{
+  constexpr std::array<char, 4>   input = { 1, 2, 3, 4  };
+  constexpr std::array<int, 4> expected = { 2, 4, 6, 8 };
+  constexpr auto actual = eve::algo::array_map(input, [](char x) -> int { return x * 2; });
+  TTS_CONSTEXPR_EXPECT(expected == actual);
+}
+
+TTS_CASE("eve.algo.array_utils array_map")
+{
+  {
+    constexpr std::array in = {1, 2, 3, 4, 5};
+    constexpr int actual = eve::algo::array_reduce(in, std::plus<>{});
```

Comment:
you can use eve::add directly, so we don't need to use `<functional>`

Reply:
What's wrong with using functional?

Reply:
well, compile time and all that ;)
It's not a big deal but I think it could be good to check it also works with our own callables.

Reply:
Can't do because then the `eve::add` and `eve::min` are not constexpr. And these functions seems like they should be constexpr friendly.

I'm also not sure about compile time at all.